### PR TITLE
transfer: Disallow a forwarding object specified with zero hops and a memo

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

disallows a forwarding object with zero hops having a memo set.

this should not change the flow throughout the code since we dont interact with the forwarding object at the time if no hops are specified.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/transfer/types/forwarding.go [hunk 1]
```diff
@@ -22,6 +22,14 @@ func (fi Forwarding) Validate() error {
 		return errorsmod.Wrapf(ErrInvalidForwarding, "number of hops in forwarding path cannot exceed %d", MaximumNumberOfForwardingHops)
 	}
 
+	if len(fi.Memo) > MaximumMemoLength {
+		return errorsmod.Wrapf(ErrInvalidMemo, "memo length cannot exceed %d", MaximumMemoLength)
+	}
+
+	if len(fi.Hops) == 0 && fi.Memo != "" {
+		return errorsmod.Wrap(ErrInvalidForwarding, "memo specified when forwarding hops are zero")
```
Comment (crodriguezvega) [inline modules/apps/transfer/types/forwarding.go:30]: nit:

```suggestion
		return errorsmod.Wrap(ErrInvalidForwarding, "memo specified when forwarding hops is empty")
```

File: modules/apps/transfer/types/forwarding_test.go [hunk 2]
```diff
@@ -56,6 +56,11 @@ func TestForwarding_Validate(t *testing.T) {
 			types.NewForwarding(ibctesting.GenerateString(types.MaximumMemoLength+1), validHop),
 			types.ErrInvalidMemo,
 		},
+		{
+			"invalid forwarding with zero hops and specified memo",
```
Comment (crodriguezvega) [inline modules/apps/transfer/types/forwarding_test.go:60]:
```suggestion
			"invalid forwarding with empty hops and specified memo",
```

**Review by crodriguezvega — APPROVED** [review]
Quick turnaround, @DimitrisJim!

**Review by gjermundgaraba — APPROVED** [review]
Great work! Do we also need to add this validation to FungibleTokenPacketDataV2?

**Pull request merged**